src/main/java/at/hannibal2/skyhanni/api/HotmAPI.kt

```diff
@@ -0,0 +1,86 @@
+package at.hannibal2.skyhanni.api
+
+import at.hannibal2.skyhanni.data.HotmData
+import at.hannibal2.skyhanni.data.ProfileStorageData
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.ItemCategory
+import at.hannibal2.skyhanni.utils.ItemUtils.getItemCategoryOrNull
+import at.hannibal2.skyhanni.utils.NEUInternalName.Companion.asInternalName
+import at.hannibal2.skyhanni.utils.SkyBlockItemModifierUtils.getDrillUpgrades
+import at.hannibal2.skyhanni.utils.StringUtils.firstLetterUppercase
+import at.hannibal2.skyhanni.utils.TimeLimitedCache
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraft.item.ItemStack
+import kotlin.time.Duration.Companion.seconds
+
+object HotmAPI {
+
+    fun copyCurrentTree() = HotmData.storage?.deepCopy()
+
+    val activeMiningAbility get() = HotmData.abilities.firstOrNull { it.enabled }
+
+    private val blueGoblinEgg = "GOBLIN_EGG_BLUE".asInternalName()
+
+    private val blueEggCache = TimeLimitedCache<ItemStack, Boolean>(10.0.seconds)
+    val isBlueEggActive
+        get() = InventoryUtils.getItemInHand()?.let {
+            blueEggCache.getOrPut(it) {
+                it.getItemCategoryOrNull() == ItemCategory.DRILL && it.getDrillUpgrades()
+                    ?.contains(blueGoblinEgg) == true
+            }
+        } == true
```

Comment:
you're checking for the blue goblin egg instead of the omelette, see https://discord.com/channels/997079228510117908/997079229302837269/1233542013828333581

Reply:
thx, fixed it.